src/main/java/com/contrast/labs/ai/mcp/contrast/RouteCoverageService.java

```diff
@@ -87,6 +90,17 @@ public RouteCoverageResponse getRouteCoverage(
       throw new IllegalArgumentException(errorMsg);
     }
 
+    // Warn if mutually exclusive parameters are both provided
+    if (Boolean.TRUE.equals(useLatestSession) && StringUtils.hasText(sessionMetadataName)) {
+      log.atWarn()
```

Comment:
Why the spring structured logging use here but not other log statements? Just to get the key/value logging?